src/content/docs/en/reference/integrations-reference.mdx

```diff
@@ -368,6 +372,42 @@ The **`stage`** denotes how this script (the `content`) should be inserted. Some
     injectScript('page-ssr', 'import "global-styles.css";');
     ```
 
+#### `injectDts`
+
+**Type:** `(dts: { name: string; content: string }) => void;`
+
+A callback function that injects a `.d.ts` file inside the user's `.astro` directory. The file will be referenced for you in the user's project, no need to create a manual reference in `src/env.d.ts`.
+
+This is useful for generating types based on user content, virtual modules etc.
+
+```js
+injectDts({
+  name: 'my-integration.d.ts',
+  content:
+    `declare module "virtual:my-integration/config" {
+      export const color: string;
+    }`
+})
+```
+
+:::note
+`astro.d.ts` and `types.d.ts` are reserved names.
+:::
+
+#### `codegenDir`
+
+**Type:** `URL`
+
+The absolute path to the `.astro` directory.
```

Comment:
Does it have to be absolute or can it be relative to the project root?
If I want it to be hidden inside of `node_modules` like Vite does with `node_modules/.vite` can I set `./node_modules/.astro`?

Or is it absolute within the project, like Vite imports, so it would be `/node_modules/.astro`?